opm/simulators/wells/StandardWell_impl.hpp

```diff
@@ -2070,20 +2070,25 @@ namespace Opm
                     }
                 } else {
                     assert(this->isProducer());
-                    // Using the preferred phase to decide the mix initialization.
-                    switch (this->wellEcl().getPreferredPhase()) {
-                    case Phase::OIL:
-                        mix[FluidSystem::oilCompIdx] = 1.0;
-                        break;
-                    case Phase::GAS:
-                        mix[FluidSystem::gasCompIdx] = 1.0;
-                        break;
-                    case Phase::WATER:
-                        mix[FluidSystem::waterCompIdx] = 1.0;
-                        break;
-                    default:
-                        // No others supported.
-                        break;
+                    // For the frist perforation without flow we use the preferred phase to decide the mix initialization.
+                    if (perf == 0) { //
+                        switch (this->wellEcl().getPreferredPhase()) {
+                        case Phase::OIL:
+                            mix[FluidSystem::oilCompIdx] = 1.0;
+                            break;
+                        case Phase::GAS:
+                            mix[FluidSystem::gasCompIdx] = 1.0;
+                            break;
+                        case Phase::WATER:
+                            mix[FluidSystem::waterCompIdx] = 1.0;
+                            break;
+                        default:
+                            // No others supported.
+                            break;
+                        }
+                    // For the rest of the perforation without flow we use mix from the above perforation.
+                    } else {
+                        mix = x;
```

Comment:
It might make no big difference and `tot_surf_rate == 0.` probably not happen very often.  Considering the flow direction, should not the perforation below be used here?   Following this line, should we do the iteration from bottom to up instead of from up to bottom?  I am not very familiar with this part of the code, so it is just some questions here.

Reply:
See comment above. The tot_surf_rate is accumulated from the bottom. If tot_surf_rate is zero for a perforation most probably all underlying perforations have zero rate. i.e. we can not use the mixture from the bottom. The logic of using the mixture from the top is that if fluid is not entering from the reservoir the well bore will fill still be filled up with something. And that something must come from the top.